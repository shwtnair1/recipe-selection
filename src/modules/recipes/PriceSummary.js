import React from 'react';
import Box from '../../components/Box';
import Flex from '../../components/Flex';
import Text from '../../components/Text';
import { parseRawPrice } from './price';

let hrStyle={
  borderTopColor:'#E4E4E4',
  borderTopWidth:'1px',
  borderTopStyle:'solid'
};
// Create PriceSummary user interface
const PriceSummary = ({ summary, totalPrice }) => {
  return(
  <Box width={['290px', '450px']} padding='16px'>
      {summary.items.map((item)=>{
        return<Flex key={item.id} justifyContent='space-between' marginBottom='8px' fontSize='16px' lineHeight='27px'>
            <Text fontWeight={400}>{item.name} {item.selected > 1 ? 'x '+item.selected:''}</Text>
            <Text>{parseRawPrice(item.price)}</Text>
          </Flex>
      })}
      <Flex justifyContent='space-between' marginTop='8px' fontSize='16px' lineHeight='27px'>
            <Text fontWeight={400} >Shipping</Text>
            <Text>{parseRawPrice(summary.shippingPrice)}</Text>
      </Flex>
      <hr style={hrStyle}/>
      <Flex justifyContent='space-between' marginTop='8px' fontSize='16px' lineHeight='27px'>
            <Text fontWeight={600}>Total</Text>
            <Text fontWeight={600}>{parseRawPrice(summary.total)}</Text>
      </Flex>
  </Box>
)
};

export default PriceSummary;
