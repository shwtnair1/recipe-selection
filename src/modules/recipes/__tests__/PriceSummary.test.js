import React from 'react';
import PriceSummary from '../PriceSummary';
import Enzyme, { shallow,mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { parseRawPrice } from '../price';
import Text from '../../../components/Text';

Enzyme.configure({ adapter: new Adapter() });

const summary={
    items:[{id:1,name:'test1',selected:1,price:1719},{id:2,name:'test2',selected:3,price:3535}],
    shippingPrice:2525,
}

const totalPrice='$77.79';

describe('PriceSummary',()=>{
  it("renders",()=>{
      const wrapper = shallow(<PriceSummary summary={summary} totalPrice={totalPrice}/>) ;
    expect(wrapper).toMatchSnapshot();
  })

  //Displays shipping Price and total price as $0.00
  it("renders without crashing, when there is no summary/totalPrice passed",()=>{
    const wrapper = shallow(<PriceSummary />) ;
  expect(wrapper).toMatchSnapshot();

  expect(wrapper.find({"data-testid":"item-list"})).toHaveLength(0);

  expect(wrapper.find({"data-testid":"shipping-price"})).toHaveLength(1);
  let children = wrapper.find({"data-testid":"shipping-price"}).find(Text);
  expect(children.at(1).text()).toBe(parseRawPrice(0));
  
  expect(wrapper.find({"data-testid":"total-price"})).toHaveLength(1);
  children = wrapper.find({"data-testid":"total-price"}).find(Text);
  expect(children.at(1).text()).toBe(parseRawPrice(0));
})

it("renders when there is no summary Items passed",()=>{
    const wrapper = shallow(<PriceSummary summary={{shippingPrice:0}} totalPrice={0}/>) ;
  expect(wrapper.find({"data-testid":"item-list"})).toHaveLength(0);
})

it("renders when there is empty summary Items passed",()=>{
    const wrapper = shallow(<PriceSummary summary={{items:[],shippingPrice:0}} totalPrice={0}/>) ;
  expect(wrapper.find({"data-testid":"item-list"})).toHaveLength(0);
})

it("renders price summary details correctly",()=>{
    const wrapper = shallow(<PriceSummary summary={summary} totalPrice={totalPrice}/>) ;
  expect(wrapper.find({"data-testid":"item-list"})).toHaveLength(2);
  expect(wrapper.find({"data-testid":"shipping-price"})).toHaveLength(1);
  expect(wrapper.find({"data-testid":"total-price"})).toHaveLength(1);
})

it("displays price details with correct parsing",()=>{
    const wrapper = shallow(<PriceSummary summary={summary} totalPrice={totalPrice}/>) ;
    let children = wrapper.find({"data-testid":"shipping-price"}).find(Text);
    expect(children.at(1).text()).toBe(parseRawPrice(summary.shippingPrice));
    children = wrapper.find({"data-testid":"item-list"}).find(Text);
    expect(children.at(1).text()).toBe(parseRawPrice(summary.items[0].price));
})

it("displays item when selected quantity is more than 1",()=>{
    const wrapper = shallow(<PriceSummary summary={summary} totalPrice={totalPrice}/>) ;
    let children = wrapper.find({"data-testid":"item-list"}).find(Text);
    expect(children.at(0).text()).toBe(summary.items[0].name);
    expect(children.at(2).text()).toBe(`${summary.items[1].name} x ${summary.items[1].selected}`);
})
});