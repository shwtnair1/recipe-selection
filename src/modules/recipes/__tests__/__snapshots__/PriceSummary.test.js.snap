// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceSummary renders 1`] = `ShallowWrapper {}`;

exports[`PriceSummary renders when there is no summary/totalPrice passed 1`] = `ShallowWrapper {}`;
